spec/support/summarise_matcher.rb

```diff
@@ -0,0 +1,41 @@
+RSpec::Matchers.define :summarise do |expected|
+  match do |_actual|
+    if expected[:action]
+      key_html && value_html && action_link && action_link[:href] == expected[:action][:href]
+    else
+      key_html && value_html
+    end
+  end
+
+  failure_message do |actual|
+    if !key_html
+      "Could not find the key ‘#{expected[:key]}’ within\n\n #{actual}"
+    elsif !value_html
+      "Could not find the value ‘#{expected[:value]}’ within \n\n#{row_html.native.to_html}"
+    elsif !action_link
+      "Could not find the link ‘#{expected[:action][:text]}’ within \n\n#{row_html.native.to_html}"
+    else
+      "Expected link href to be #{expected[:action][:href]}, was #{action_link[:href]}"
+    end
+  end
+
+  def html
+    @html ||= Capybara::Node::Simple.new(actual)
+  end
+
+  def key_html
+    @key_html ||= html.all('dt.govuk-summary-list__key', text: expected[:key]).first
+  end
+
+  def row_html
+    @row_html ||= key_html.ancestor('.govuk-summary-list__row')
+  end
+
+  def value_html
+    @value_html ||= row_html.all('dd.govuk-summary-list__value', text: expected[:value]).first
+  end
+
+  def action_link
+    @action_link ||= row_html.all(:link, expected[:action][:text]).first
```

Comment:
Just wondering why the `options` param here is a string but it's a `{ text: <value> }` hash in the other cases above? From the docs it looks like a string defaults to `exact_text`. Would it be best to do an `exact_text` lookup for all the above?

Reply:
@stevehook good spot! This inconsistency was accidental... 🤦 

I've switched them all to do an `exact_text` match now. Also updated the failure message to be slightly more readable if the value text doesn't match. Let me know what you think!